src/bundle.c

```diff
@@ -1625,6 +1625,47 @@ gboolean check_bundle_payload(RaucBundle *bundle, GError **error)
 	return res;
 }
 
+gboolean extract_signature(RaucBundle *bundle, const gchar *outputsig, GError **error)
+{
+	GError *ierror = NULL;
+	g_autoptr(GFile) sigfile = NULL;
+	g_autoptr(GFileIOStream) sigstream = NULL;
+	GOutputStream *sigoutstream = NULL; /* owned by the signature stream */
+	gboolean res = FALSE;
+
+	g_return_val_if_fail(bundle != NULL, FALSE);
+	g_return_val_if_fail(outputsig != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	r_context_begin_step("extract_signature", "Extracting bundle signature", 0);
+
+	sigfile = g_file_new_for_path(outputsig);
+	sigstream = g_file_create_readwrite(sigfile, G_FILE_CREATE_PRIVATE, NULL, &ierror);
+
+	if (sigstream == NULL) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"failed to create file to store signature: ");
+		goto out;
+	}
+
+	sigoutstream = g_io_stream_get_output_stream(G_IO_STREAM(sigstream));
```

Comment:
You could use `g_file_create()` instead, which directly produces a GFileOutputStream (which can be casted to  a `GOutputStream`).

Reply:
Yes for sure ! My English is not always very good

Reply:
The second part `rauc replace-signature` will coming soon 😄